File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.java
class Solution {
    public String solution(String s) {
        String answer = "";
        
        StringBuilder sb = new StringBuilder();
        
        int index = 0;
        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            if (ch == ' ') {
                sb.append(" ");
                index = 0;
            }
            else {
                sb.append(index%2==0 ? Character.toUpperCase(ch) : Character.toLowerCase(ch));
                index++;
            }
        }
        
        answer = sb.toString();
        
        return answer;
    }
}

File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.js
function solution(s) {
    let answer = '';
    
    let index = 0;
    for (let i = 0; i < s.length; i++) {
        let c = s[i];
        if (c == ' ') {
            answer += ' ';
            index = 0;
        } else {
            if (index%2 == 1)
                answer += c.toLowerCase();
            else
                answer += c.toUpperCase();
            
            index++;
        }
    }
    
    return answer;
}